distributed/scheduler.py

```diff
@@ -2126,6 +2126,18 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
             n_workers: Py_ssize_t = len(worker_pool_dv)
             if n_workers < 20:  # smart but linear in small case
                 ws = min(worker_pool.values(), key=operator.attrgetter("occupancy"))
+                if ws.occupancy == 0:  # special case to use round-robin
+                    wp_vals = worker_pool.values()
+                    start = self._n_tasks % n_workers
+                    for i in range(start, n_workers):
+                        if wp_vals[i].occupancy == 0:
+                            ws = wp_vals[i]
+                            break
+                    else:
+                        for i in range(start):
+                            if wp_vals[i].occupancy == 0:
+                                ws = wp_vals[i]
```

Comment:
Similar suggestion to the code above though I guess whether this is needed in 2 places or 1 depends on the discussion about these loops. Anyways just included this for completeness. Hopefully this gives an idea of how one can approach these things

```suggestion
                            wp_i = wp_vals[i]
                            if wp_i._occupancy == 0:
                                ws = wp_i
```